go/runtime/host/protocol/connection.go

```diff
@@ -321,12 +325,22 @@ func (c *connection) workerOutgoing() {
 	for {
 		select {
 		case msg := <-c.outCh:
+			if err := c.conn.SetWriteDeadline(time.Now().Add(connWriteTimeout)); err != nil {
+				c.logger.Error("error setting connection deadline",
+					"err", err,
+				)
+			}
 			// Outgoing message, send it.
 			if err := c.codec.Write(msg); err != nil {
 				c.logger.Error("error while sending message",
 					"err", err,
 				)
 			}
+			if err := c.conn.SetWriteDeadline(time.Time{}); err != nil {
```

Comment:
Since `Write` is only ever called here, do we need to reset the deadline?

Reply:
need no, but seemed like a reasonable thing to do (specially since connection is passed into this struct and not created here).

Reply:
Ok.